src/js/stores/MesosStateStore.js

```diff
@@ -141,7 +141,7 @@ class MesosStateStore extends GetSetBaseStore {
   }
 
   getTasksFromNodeID(nodeID) {
-    const { tasks, frameworks } = this.get("lastMesosState");
+    const { tasks = [], frameworks = [] } = this.get("lastMesosState") || {};
```

Comment:
Here I saw it throw an error once because `framework` was undefined on line 150. This causes line 158 to throw an exception.

Reply:
This seems to be quite hard to unit test 😞 I tried for 20 mins, but failed to come up with a test that fails.

Reply:
You can try testing it with fixtures. Not sure how it's gonna work with the stream though... Or just good old hardcode :trollface: